### PR TITLE
Test API using behave

### PR DESCRIPTION
**Pull request by richdouglasevans**
```bash
$ behave
Feature: Give me some sentences # features\sentences.feature:1
  We want sentences from the API.
  Scenario: We can request sentences for a particular feature  # features\sentences.feature:5
    Given these sentences exist                                # features\steps\sentences.py:10
      | sentence              | features   |
      | Adam's car is lovely. | apostrophe |
    When we request sentences with the 'apostrophe' feature    # features\steps\sentences.py:15
    Then these sentences are returned                          # features\steps\sentences.py:25
      | sentence              |
      | Adam's car is lovely. |

  Scenario: We can limit the number of sentences we receive    # features\sentences.feature:16
    Given these sentences exist                                # features\steps\sentences.py:10
      | sentence                                          | features   |
      | Adam's car is lovely.                             | apostrophe |
      | Adam's passion for artisanal butter is legendary. | apostrophe |
    When we request one sentence with the 'apostrophe' feature # features\steps\sentences.py:20
    Then these sentences are returned                          # features\steps\sentences.py:25
      | sentence              |
      | Adam's car is lovely. |

1 feature passed, 0 failed, 0 skipped
2 scenarios passed, 0 failed, 0 skipped
6 steps passed, 0 failed, 0 skipped, 0 undefined
Took 0m0.039s
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,14 @@ $ python wsgi.py
...
+
 ```bash
 $ nosetests
 ```
+
+Functional tests:
+
+```bash
+$ behave
+```
```
Comment (richdouglasevans) [inline README.md]: Note: for these tests you _must_ have a local MongoDB instance running.

File: features/environment.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from krump import create_app, required_value
+
+
+# noinspection PyUnusedLocal
+def before_feature(context, feature):
+    app = create_app()
+    app.testing = True
+
+    setup_db(app, context)
+    setup_api(app, context)
```
Comment (richdouglasevans) [inline features/environment.py]: Before any features are run we just start up the Flask application and create a few helper classes that we assign to the context to make steps nice and thin.

File: features/sentences.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	Scenario: We can limit the number of sentences we receive
+
+		Given these sentences exist
+			| sentence                                          | features   |
+			| Adam's car is lovely.                             | apostrophe |
+			| Adam's passion for artisanal butter is legendary. | apostrophe |
+		When we request one sentence with the 'apostrophe' feature
+		Then these sentences are returned
+			| sentence              |
+			| Adam's car is lovely. |
```
Comment (richdouglasevans) [inline features/sentences.feature]: We can add more scenarios as we add more functionality to the API.

Outstanding:

* `max_words`

**Comment by lpmi-13** [comment]
I'm assuming the second test of apostrophes is meant merely as an example template, since it should be retuning two sentences instead of just one. Merging away!

File: features/sentences.feature (hunk 3 above)
Comment (lpmi-13) [inline features/sentences.feature]: I'll see if I can get a working implementation of the max_words, and possibly other stuff as well

Comment (richdouglasevans) [inline features/sentences.feature]: 👍 Go Adam, go! 😄

**Comment by richdouglasevans** [comment]
> I'm assuming the second test of apostrophes is meant merely as an example template, since it should be retuning two sentences instead of just one.

No. If you examine the `When` clause of that second scenario you can see that we are requesting just _one_ sentence.

```gherkin
Scenario: We can limit the number of sentences we receive

	Given these sentences exist
		| sentence                                          | features   |
		| Adam's car is lovely.                             | apostrophe |
		| Adam's passion for artisanal butter is legendary. | apostrophe |
	When we request one sentence with the 'apostrophe' feature
	Then these sentences are returned
		| sentence              |
		| Adam's car is lovely. |
```

There's no explicit sort order so this test is a little brittle: it only passes because the first sentence that was added happens to be the first sentence that is retrieved.

**Comment by richdouglasevans** [comment]
Oh, and please do change the references to _"artisanal butter"_ if you don't dig them. I tend to pepper random strings in test code with whatever random thought pops into mind.

Uh. Not that you or artisanal butter are on my mind. Oh, now my mind cannot unsee...

**Pull request merged**